src/Build/ElementLocation/XmlDocumentWithLocation.cs

```diff
@@ -115,7 +115,20 @@ private enum ReadOnlyLoadFlags
         internal string FullPath
         {
             get { return _fullPath; }
-            set { _fullPath = value; }
+            set
+            {
+                if (!FullPathHasChanged && !String.Equals(_fullPath, value, StringComparison.OrdinalIgnoreCase))
+                {
+                    FullPathHasChanged = true;
+                }
+                _fullPath = value;
+            }
+        }
+
+        internal bool FullPathHasChanged
```

Comment:
It seems to me that once you "SaveAs" you pay the cost of checking the full path forever?

Reply:
At the moment, you pay the price always.  With my change, projects loaded from disk no longer create a new ElementLocation every time.  In-memory projects that are saved will pay the cost once.